container/go/pkg/compat/ociToLegacy.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////
+// This utility works with the new_container_pull and new_container_load targets
+// to generate the appropriate pseudo-intermediate format that is compatible
+// with the rules_docker container_import rule.
+
+package compat
+
+import (
+	ospkg "os"
+	"path"
+	"strconv"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/pkg/errors"
+)
+
+// Extension for layers and config files that are made symlinks
+const targzExt = ".tar.gz"
+const configExt = "config.json"
+
+// generateSymlinks safely generates a symbolic link from src to dst.
+func generateSymlinks(src, dst string) error {
+	if _, err := ospkg.Stat(src); err != nil {
+		return errors.Wrapf(err, "source file does not exist at %s", src)
+	}
+
+	if _, err := ospkg.Lstat(dst); err == nil {
+		if err = ospkg.Remove(dst); err != nil {
+			return errors.Wrapf(err, "failed to remove existing file at %s", dst)
+		}
+	}
+	if err := ospkg.Symlink(src, dst); err != nil {
+		return errors.Wrapf(err, "failed to create symbolic link from %s to %s", src, dst)
```

Comment:
the symbolic link is actually being created from dst to src here. ie. dst is the symlink and reading it will result in reading the contents of src. See my comment above to fix this message.